plugins/automatic_alignment.py

```diff
@@ -0,0 +1,367 @@
+        return data
+
+    def open_image(self, dlg):
+        # Find the available formats (and corresponding extensions)
+        formats_to_ext = dataio.get_available_formats(os.O_RDONLY)
+        config = get_acqui_conf()
+        path = config.last_path
+
+        wildcards, formats = guiutil.formats_to_wildcards(formats_to_ext, include_all=True)
+        # TODO dlg.pnl_desc?
+        dialog = wx.FileDialog(dlg.pnl_desc,
+                               message="Choose a file to load",
+                               defaultDir=path,
+                               defaultFile="",
+                               style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST,
+                               wildcard=wildcards)
+
+        # Show the dialog and check whether is was accepted or cancelled
+        if dialog.ShowModal() != wx.ID_OK:
+            dlg.Destroy()
+            return
+
+        # Detect the format to use
+        filename = dialog.GetPath()
+
+        data = open_acquisition(filename)[0]
+        try:
+            data = self._ensureGrayscale(data)
+        except ValueError, exception:
+            box = wx.MessageDialog(dlg, str(exception), "Exit", wx.OK | wx.ICON_STOP)
+            box.ShowModal()
+            box.Destroy()
+            dlg.Destroy()
+            return
+
+        s = stream.StaticSEMStream("TEM stream", data)
+        tem_projection = AlignmentProjection(s)
+        crop = (self.crop_top.value, self.crop_bottom.value,\
+                self.crop_left.value, self.crop_right.value)
+        flip = (self.flip_x.value, self.flip_y.value)
+        tem_projection.setPreprocessingParams(self.invert.value, flip, crop, self.blur.value, True)
+        dlg.addStream(tem_projection, 0)
+        self._temStream = tem_projection
+
+    def align(self, dlg):
+        ''' Executes the alignment. If the alignment is successful, the aligned stream is
+            added to the main window. If not, an error message is shown.
+        dlg (AlignmentAcquisitionDialog): The plugin dialog
+        '''
+        crop = (self.crop_top.value, self.crop_bottom.value,\
+                self.crop_left.value, self.crop_right.value)
+        flip = (self.flip_x.value, self.flip_y.value)
+        tem_img = preprocess(self._temStream.raw[0], self.invert.value, flip, crop,
+                self.blur.value, True)
+        sem_img = preprocess(self._semStream.raw[0], False, (False, False), (0, 0, 0, 0), 5, True)
+        try:
+            tmat, kp_tem, kp_sem = keypoint.FindTransform(tem_img, sem_img)
+        except ValueError as exception:
+            box = wx.MessageDialog(dlg, str(exception), "Exit", wx.OK | wx.ICON_STOP)
+            box.ShowModal()
+            box.Destroy()
+            dlg.Destroy()
+            return
+
+        # get the metadata corresponding to the transformation
+        transf_md = get_img_transformation_md(tmat, tem_img, sem_img)
+
+        orig_tem_ps = tem_img.metadata.get(model.MD_PIXEL_SIZE, (1e-9, 1e-9))
```

Comment:
Move all this trickery of metadata computation into conversion.get_img_transformation_md().